File: 백준/Bronze/1978. 소수 찾기/소수 찾기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		//  hasMoreTokens(): StringTokenizer 자체가 토큰이 남아있는지의 여부를 true, false 로 반환시켜줌
		br.readLine();	// 굳이 N을 쓸 필요 없음
		int cnt = 0;	// 소수 개수
		
		StringTokenizer st = new StringTokenizer(br.readLine(), " ");
		
		while(st.hasMoreTokens()) {	
	        
			// 소수인 경우 true, 아닌 경우 false   
			boolean isPrime = true;
			
			int num = Integer.parseInt(st.nextToken());
			
			if(num == 1) {
				continue;
			}
			for(int i = 2; i <= Math.sqrt(num); i++) {
				if(num % i == 0) {
					isPrime = false;
					break;
				}
			}
			if(isPrime) {
				cnt++;
			}
		}
		System.out.println(cnt);		
	}
}

File: 백준/Bronze/1978. 소수 찾기/소수 찾기.cc
#include <iostream>
using namespace std;

int main() 
{
	int N, num;
	int div = 0;    // 약수의 개수
	int cnt = 0;    // 소수의 개수
	
	cin >> N;
	
	for (int i = 0; i < N; i++)
	{
		cin >> num;
		
		if (num == 1)
			continue;
			
		for (int j = 1; j <= num; j++)
		{
			if (num % j == 0)
			{
				div++;
                
                // 약수의 개수가 이미 2개를 초과하면 계속 반복할 필요 없음
				if (div > 2)
					break;
			}
		}
		
        // 약수의 개수가 2개라는 것은 소수라는 뜻
		if (div == 2)
		    cnt++;
		
		div = 0;
	}
	
	cout << cnt;
}